0_From_Code_First_To_First_Principles/1_code_first/assignment/src/modules/user/commands/use-cases/create-user/create-user.use-case.ts

```diff
@@ -0,0 +1,30 @@
+import { Log } from "../../../../../libs/log/log";
+import { UserEntity } from "../../../entities/user.entity";
+import { generateRandomPassword } from "../../../helpers/generate-random-password";
+import { IUserRepository } from "../../../repository/contracts/user.repository.interface";
+import { CreateUserCommand } from "./contracts/create-user.command";
+import { ICreateUserUseCase } from "./contracts/create-user.use-case.interface";
+
+export class CreateUserUseCase implements ICreateUserUseCase {
+  constructor(private readonly repository: () => Promise<IUserRepository>) {}
+
+  @Log({
+    startMessage: "Running CreateUser with input:",
+    logInputValues: true,
+    endMessage: "Done, result:",
+    logReturnValue: true,
+  })
```

Comment:
I really like how you've handled the logs

Reply:
Thank you! :)